parallel-crypto3/libs/parallel-zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/lookup_argument.hpp

```diff
@@ -343,19 +358,32 @@ namespace nil {
                                         polynomial_dfs_variable_type::column_type::selector);
 
                             lookup_tags.insert(lookup_tag);
+                            std::vector<std::vector<expression_evaluator_registration>>
+                                registrationss;
                             for (size_t o_id = 0; o_id < l_table.lookup_options.size(); ++o_id) {
+                                std::vector<expression_evaluator_registration>
+                                    registrations;
                                 for (std::size_t i = 0; i < l_table.columns_number; i++) {
-                                    expressions.push_back(lookup_tag * l_table.lookup_options[o_id][i]);
+                                    auto expr =
+                                        lookup_tag * l_table.lookup_options[o_id][i];
+                                    registrations.push_back(
+                                        _central_expr_evaluator.register_expression(
+                                            std::move(expr)));
                                 }
+                                registrationss.push_back(std::move(registrations));
                             }
+                            registrationsss.push_back(std::move(registrationss));
```

Comment:
You could consider to rename 'registrationsss' to something else. Maybe "all_gates_registrations" or something like that.